packages/next/next-server/server/next-server.ts

```diff
@@ -1039,24 +1040,67 @@ export default class Server {
       this.dynamicRoutes = this.getDynamicRoutes()
     }
 
+    const rewrites = {
+      beforeFiles,
+      afterFiles,
+      fallback,
+    }
+    const clientHeaders = this.getClientHeaders(rewrites)
+
     return {
       headers,
       fsRoutes,
-      rewrites: {
-        beforeFiles,
-        afterFiles,
-        fallback,
-      },
+      rewrites,
       redirects,
       catchAllRoute,
       useFileSystemPublicRoutes,
       dynamicRoutes: this.dynamicRoutes,
       basePath: this.nextConfig.basePath,
       pageChecker: this.hasPage.bind(this),
       locales: this.nextConfig.i18n?.locales || [],
+      clientHeaders,
     }
   }
 
+  protected getClientHeaders(rewrites: Router['rewrites']): string[] {
+    const BLOCK_HEADERS = ['authorization', 'host']
```

Comment:
Not sure if this is failsafe enough? What do y'all think?

Reply:
Don't think anyone would be using a header like this to do rewrites anyway, right?

Reply:
no, I don't think so